### DISCUSSION TITLE
Change AMD-APP to PoCL in openSUSE Tumbleweed

### DISCUSSION THREAD

**Discussion by RafaelLinux** [pocl/pocl#1536] [Q&A]
Please, someone with the right steps to get PoCL working in openSUSE Tumbleweed?. I gave up after several tries and no getting to change.

Thanks

**Comment by pjaaskel** [comment]
I'd assume basic steps as instructed in the README should work. Which problems did you encouter?

**Comment by RafaelLinux** [comment]
I didn't find any about AMD GPUs, only about NVIDIA in README.

**Comment by pjaaskel** [comment]
OK. Your initial question didn't either mention AMD GPUs. I thought you referred to the AMD OpenCL APPs. AMD GPUs are not supported currently, unless perhaps some older one running on the pure HSA stack.

**Comment by jansol** [comment]
You may have some luck with the vulkan backend, if the card is new enough to support vulkan. Just beware that that is still in an experimental state.

**Comment by RafaelLinux** [comment]
I mentioned AMD in the thread title, maybe that's why you didn't notice it at first. 

In any case, I'm still just as confused after your reply. I just asked here because a Tumbleweed user [posted](https://forums.opensuse.org/t/amd-gpu-rx-6700-xt-support-in-tumbleweed/177772) the steps he took to get AMD OpenCL on Tumbleweed. Interestingly, what he has finally installed is precisely PoCL with OpenCL 3.0 . The curious thing about this is not only that he has PoCL with AMD GPUs, but that in his installation steps, PoCL does not appear anywhere.

Any suggestion about how to install PoCL in Tumbleweed with an AMD GPU?

**Reply by jansol** [reply]
That post talks about using AMD's ROCm to provide OpenCL and HIP drivers. PoCL looks to be a red herring there. It appears it simply happens to be installed and the user seems to have clipped the wrong part from `clinfo` output. As for why OpenCL stopped working when removing PoCL, that may have been caused by something like the PoCL package pulling in the ICD loader (`ocl-icd-libopencl1` on debian-likes) as a dependency that got autoremoved when removing PoCL.

PoCL can be installed side by side with ROCm and can be used for AMD CPU devices at the same time as ROCm is used for AMD GPUs thanks to the ICD loader and the platform mechanism. But that won't make the AMD GPU show up in PoCL.

**Comment by RafaelLinux** [comment]
Thank you for shedding some light on the strangeness of the data displayed by ‘clinfo’ on that user's computer.
As I understand there is no way to make PoCL run on AMD GPU's, I close this thread.
